internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -1705,27 +1656,41 @@ func BootstrapEtcd(seq runtime.Sequence, data interface{}) (runtime.TaskExecutio
 			}
 		}
 
-		// Since etcd has already attempted to start, we must delete the data. If
-		// we don't, then an initial cluster state of "new" will fail.
-		dir, err := os.Open(constants.EtcdDataPath)
-		if err != nil {
-			return err
-		}
+		if err = func() error {
```

Comment:
this fixes a bug in `etcd` bootstrap: if `etcd` never started before `talosctl bootstrap` is issued, etcd data directory might not exist yet